dstyle.dd

```diff
@@ -421,9 +421,9 @@ bool isPositive(int number)
 }
 ---
     $(LI Text in sections (e.g. `Params:`, `Returns:`, `See_Also`) should be indented by one level if it spans more than the line of the section.)
-    $(LI Documentation comments should not use more than two stars `/**` in the header line.)
-    $(LI Block comments (`/**`) should be used - not nesting block comments (`/++`))
-    $(LI Global functions shouldn't indent their documentation block nor use stars as indentation.)
+    $(LI Documentation comments should not use more than two stars `/**` or two pluses `/++` in the header line.)
+    $(LI Either Block comments (`/**`) or nesting block comments (`/++`) should be used except when the ddoc comment is a ditto comment such as `/// Ditto`)
+    $(LI Documentation comments should not have leading stars or zeroes on each line.)
```

Comment:
Anyhow I'm not sure whether I messed up the wording.
I saw a couple of Phobos functions doing this:

```d
/**
    Sorts a range by multiple keys. The call $(D multiSort!("a.id < b.id",
    "a.date > b.date")(r)) sorts the range `r` by `id` ascending,
    and sorts elements that have the same `id` by `date`
    descending. Such a call is equivalent to $(D sort!"a.id != b.id ? a.id
    < b.id : a.date > b.date"(r)), but `multiSort` is faster because it
    does fewer comparisons (in addition to being more convenient).
    
    Returns:
        The initial range wrapped as a `SortedRange` with its predicates
        converted to an equivalent single predicate.
*/
```

in comparison to:

```d
/**
Sorts a range by multiple keys. The call $(D multiSort!("a.id < b.id",
"a.date > b.date")(r)) sorts the range `r` by `id` ascending,
and sorts elements that have the same `id` by `date`
descending. Such a call is equivalent to $(D sort!"a.id != b.id ? a.id
< b.id : a.date > b.date"(r)), but `multiSort` is faster because it
does fewer comparisons (in addition to being more convenient).

Returns:
    The initial range wrapped as a `SortedRange` with its predicates
    converted to an equivalent single predicate.
 */
```

and of course I have observed a mixture between them too:

```d
/**
    Sorts a range by multiple keys. The call $(D multiSort!("a.id < b.id",
    "a.date > b.date")(r)) sorts the range `r` by `id` ascending,
    and sorts elements that have the same `id` by `date`
    descending. Such a call is equivalent to $(D sort!"a.id != b.id ? a.id
    < b.id : a.date > b.date"(r)), but `multiSort` is faster because it
    does fewer comparisons (in addition to being more convenient).
    
Returns:
    The initial range wrapped as a `SortedRange` with its predicates
    converted to an equivalent single predicate.
 */
```

I was trying to talk about cases like the first one where the entire documentation block is indented.

Reply:
The first case is exactly what I always do, and I expect that if you look, you'll find that that style is very common in Phobos.

Reply:
Hehe a lot of documentation styles are common in Phobos. It contains almost every permutation imaginable.
Anyhow I didn't wanted to start a war on this, but just give newcomers a better guide.